core/src/ephemeral_server/mod.rs

```diff
@@ -463,15 +463,19 @@ impl EphemeralExe {
                     EphemeralExeVersion::Fixed(version) => version,
                 };
                 let client = reqwest::Client::new();
-                let info: DownloadInfo = client
+                let resp = client
                     .get(format!(
                         "https://temporal.download/{artifact_name}/{version_name}"
                     ))
                     .query(&get_info_params)
                     .send()
-                    .await?
-                    .json()
                     .await?;
+                let info: DownloadInfo = match resp.error_for_status() {
+                    Ok(r) => r.json().await?,
+                    Err(e) => {
+                        return Err(anyhow::Error::from(e));
+                    }
+                };
```

Comment:
For patterns like:

```rs
match some_result {
  Ok(_) => ...,
  Err(e) => return Err(...),
}
```

I suggest using [.map_err(...)?](https://doc.rust-lang.org/std/result/enum.Result.html#method.map_err) instead; I find it a lot cleaner to read.

Reply:
Yeah, I also prefer that but the result type is different.

Reply:
I'd need to cast from `Result<Response>` to `anyhow::Result<PathBuf>`.
Same applies below but there's a different return type.

Reply:
Hmm.. I think I figured out a better way to write it.